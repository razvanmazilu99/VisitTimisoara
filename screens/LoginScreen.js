import { Text, View, StyleSheet, TextInput, TouchableOpacity } from "react-native";
import React from 'react';
import * as firebase from 'firebase';

export default class LoginScreen extends React.Component{
    
    constructor(props){
        super(props)

        this.state = ({
            //username: '',
            email: '',
            password: ''
        })
    }

    loginUser = (email, password) => {
        try{
            
            firebase.auth().signInWithEmailAndPassword(email,password);/*.then(function (user){
                console.log(user);
                console.log(user.email)
            })*/
             

        }catch(error){
            //console.log(error.toString())
            //alert("Error:",error)
           firebase.auth().onAuthStateChanged(user => {alert(user.email)});
        }
    }

    render(){
        return(
            <View style={styles.container}>
                <Text style={styles.loginText}>Log in</Text>

                <View>
                    <TouchableOpacity style={styles.facebookButton}>
                        <View>
                            <Text style={{fontSize: 36, fontWeight: "bold", color: 'white', left: 20}}>f</Text>
                        </View>
                        <View>
                            <Text style={styles.facebookText}>Facebook</Text>
                        </View>
                    </TouchableOpacity>
                </View>
            
                <Text style={{ color:'white', fontSize: 18, alignSelf: 'center', top: 110}}>or log in with email</Text>

                <TextInput onChangeText={(email) => this.setState({email})} placeholder="Email" style={styles.textInput}></TextInput>
                <TextInput onChangeText={(password) => this.setState({password})} placeholder="Password" secureTextEntry={true} style={[styles.textInput, {top: 150}]}></TextInput>

                <TouchableOpacity onPress={() => this.loginUser(this.state.email, this.state.password)} style={styles.loginButton}>
                    <Text style={{color: 'white', fontSize: 26, textAlign: 'center', top: 12}}>Log in</Text>
                </TouchableOpacity>

                <Text style={{color: 'white', fontSize: 18, textAlign: 'center', top: 240}}>Don't have an account?</Text>

                <TouchableOpacity onPress={() => this.props.navigation.navigate('Signup')} style={styles.signupButton}>
                    <Text style={{color: 'black', fontSize: 22, textAlign: 'center', top: 8}}>Sign up</Text>
                </TouchableOpacity>

            </View>
        ); 
    }
      
}

const styles = StyleSheet.create({
    
    container: {
        flex: 1,
        backgroundColor: '#1A1B29',
        alignItems: 'center',
    },

    loginText: {
        top: 58,
        color: 'white',
        fontSize: 48,
    },
    
    facebookButton: {
        backgroundColor: '#3552B8',
        borderRadius: 20,
        width: 152,
        height: 46,
        top: 90
    },

    facebookText: {
        color: 'white',
        fontSize: 18,
        left: 50,
        top: -35
    },

    textInput: {
        backgroundColor: 'white',
        borderRadius: 20,
        width: 337,
        height: 61,
        top: 130,
        fontSize: 26,
        paddingLeft: 25
    },

    loginButton: {
        backgroundColor: "#59B398",
        width: 337,
        height: 61,
        borderRadius: 20,
        top: 200
    },

    signupButton: {
        width: 257,
        height: 46,
        borderRadius: 20,
        backgroundColor: 'white',
        top: 260
    }
});