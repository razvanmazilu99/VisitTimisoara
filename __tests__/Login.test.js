import React from 'react';
import LoginScreen from '../screens/LoginScreen';

import { create } from 'react-test-renderer';
import renderer from 'react-test-renderer';

const tree = create(<LoginScreen />);

test('snapshot', () => {
    expect(tree).toMatchSnapshot();
});

it('login function empty email test', () => {
    let data = renderer.create(<LoginScreen />).getInstance();

    window.alert = jest.fn();
    data.loginUser('','');
    
    expect(window.alert).toHaveBeenCalledTimes(1);
    expect(window.alert).toHaveBeenCalledWith("Write your email!");
})

it('login function empty password test', () => {
    let data = renderer.create(<LoginScreen />).getInstance();

    window.alert = jest.fn();
    data.setState({email: 'username@yahoo.com'});
    data.loginUser(data.state.email,'');
    
    expect(window.alert).toHaveBeenCalledTimes(1);
    expect(window.alert).toHaveBeenCalledWith("Write your password!");
})