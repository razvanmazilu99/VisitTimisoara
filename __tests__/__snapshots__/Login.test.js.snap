// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#1A1B29",
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 48,
        "top": 58,
      }
    }
  >
    Log in
  </Text>
  <View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#3552B8",
          "borderRadius": 20,
          "height": 46,
          "opacity": 1,
          "top": 90,
          "width": 152,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 36,
              "fontWeight": "bold",
              "left": 20,
            }
          }
        >
          f
        </Text>
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "left": 50,
              "top": -35,
            }
          }
        >
          Facebook
        </Text>
      </View>
    </View>
  </View>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "white",
        "fontSize": 18,
        "top": 110,
      }
    }
  >
    or log in with email
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Email"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 20,
        "fontSize": 26,
        "height": 61,
        "paddingLeft": 25,
        "top": 130,
        "width": 337,
      }
    }
    underlineColorAndroid="transparent"
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Password"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderRadius": 20,
          "fontSize": 26,
          "height": 61,
          "paddingLeft": 25,
          "top": 130,
          "width": 337,
        },
        Object {
          "top": 150,
        },
      ]
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#59B398",
        "borderRadius": 20,
        "height": 61,
        "opacity": 1,
        "top": 200,
        "width": 337,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 26,
          "textAlign": "center",
          "top": 12,
        }
      }
    >
      Log in
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 18,
        "textAlign": "center",
        "top": 240,
      }
    }
  >
    Don't have an account?
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 20,
        "height": 46,
        "opacity": 1,
        "top": 260,
        "width": 257,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 22,
          "textAlign": "center",
          "top": 8,
        }
      }
    >
      Sign up
    </Text>
  </View>
</View>
`;
